server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/RegexMappingRoutes.java

```diff
@@ -0,0 +1,101 @@
+package org.apache.james.webadmin.routes;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.james.rrt.api.RecipientRewriteTable;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.rrt.lib.MappingSource;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.dto.AddRegexMappingRequestBodyDTO;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.JsonExtractor;
+import org.eclipse.jetty.http.HttpStatus;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+
+import static spark.Spark.halt;
+
+public class RegexMappingRoutes implements Routes {
+    public static final String BASE_PATH = "/mappings/regex";
+    private final RecipientRewriteTable recipientRewriteTable;
+    private final JsonExtractor<AddRegexMappingRequestBodyDTO> jsonExtractor;
+
+    // Constructor
+    public RegexMappingRoutes(RecipientRewriteTable recipientRewriteTable) {
+        this.recipientRewriteTable = recipientRewriteTable;
+        this.jsonExtractor = new JsonExtractor<>(AddRegexMappingRequestBodyDTO.class);
+    }
+
+    @Override
+    public String getBasePath() {
+        return BASE_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.post(getBasePath(), this::addRegexMappingRoutes );
+    }
+
+    @POST
+    @Path(BASE_PATH)
+    @ApiOperation(value = "adding address-regex mappings to RecipientRewriteTable")
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "No body created"),
+            @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid body"),
+            @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error " +
+                    "- Something went bad on the server side.")
+    })
+
+    public HaltException addRegexMappingRoutes(Request request, Response response) throws Exception {
```

Comment:
`addRegex` is enough for a method name here

Reply:
addRegexMappingRoutes meets the requirement of the ticket ("corresponding routes for JMX ADDREGEXMAPPING command") IMO.

Reply:
At the limit I would be ok with `addRegexMapping`. But the `Routes` is too much. The class is already `RegexMappingRoutes`, we know the methods are gonna be routes. And why the plural anyway? :)

Reply:
I got your point. It meets requirements, but not necessary, and maybe redundant, so I will take the name addRegexMapping. The plurals are my experience from the previous ticket. :) .